docs/specs/webhook.md

```diff
@@ -142,77 +124,71 @@ If any handler fails the operation, the operation is failed. The operation fails
+The response body of non-blocking event webhook handler is ignored.
 
-The time spent in an AFTER event delivery must not exceed 60 seconds, otherwise it would be considered as a failed delivery.
+### Future works of non-blocking events
 
-All AFTER events with registered webhook handlers are persisted into the database, with minimum retention period of 30 days.
-
-The response body of AFTER event webhook handler is ignored.
+All non-blocking events with registered webhook handlers are persisted into the database, with minimum retention period of 30 days.
 
 If any delivery failed, all deliveries will be retried after some time, regardless of whether some deliveries may have succeeded. The retry is performed with a variant of exponential back-off algorithm. If `Retry-After:` HTTP header is present in the response, the delivery will not be retried before the specific time.
 
 If the delivery keeps on failing after 3 days from the time of first attempted delivery, the event will be marked as permanently failed and will not be retried automatically.
 
-## Webhook Mutations
 
-BEFORE event webhook handler can request mutation on some fields before committing to the database.
+## Webhook Event List
+
+### Blocking Events
 
-Webhook handler cannot request mutation if the operation is failed by the handler.
+- [pre_signup](#pre_signup-admin_api_create_user)
+- [admin_api_create_user](#pre_signup-admin_api_create_user)
 
-Webhook handler request mutation in its response. For example,
+### Non-blocking Events
 
-```json
+- [user.created.user_signup](#usercreateduser_signup-usercreatedadmin_api_create_user)
+- [user.created.admin_api_create_user](#usercreateduser_signup-usercreatedadmin_api_create_user)
+- [identity.created.user_add_identity](#identitycreateduser_add_identity-identitycreatedadmin_api_add_identity)
+- [identity.created.admin_api_add_identity](#identitycreateduser_add_identity-identitycreatedadmin_api_add_identity)
+- [identity.deleted.user_remove_identity](#identitydeleteduser_remove_identity-identitydeletedadmin_api_remove_identity)
+- [identity.deleted.admin_api_remove_identity](#identitydeleteduser_remove_identity-identitydeletedadmin_api_remove_identity)
+- [identity.updated.user_update_identity](#identityupdateduser_update_identity)
+- [session.created.user_signup](#sessioncreateduser_signup-sessioncreateduser_login)
+- [session.created.user_login](#sessioncreateduser_signup-sessioncreateduser_login)
+- [session.deleted.user_revoke_session](#sessiondeleteduser_revoke_session-sessiondeleteduser_logout-sessiondeletedadmin_api_revoke_session)
+- [session.deleted.user_logout](#sessiondeleteduser_revoke_session-sessiondeleteduser_logout-sessiondeletedadmin_api_revoke_session)
+- [session.deleted.admin_api_revoke_session](#sessiondeleteduser_revoke_session-sessiondeleteduser_logout-sessiondeletedadmin_api_revoke_session)
+- [user.promoted.user_promote](#userpromoteduser_promote)
```

Comment:
Grammatically the verb "promote" is transitive verb so if we put it into full sentence, the sentence would be "The user promoted themselves" or "The user was promoted"

But it looks a bit ambiguous if we define the event name as `user.promoted.user_promote_user`.
Or it is not consistent if we use passive voice only for this event. Would `user.promoted.user_promote_themselves` a good name?

Reply:
IMO `user.promoted.user_promote_themselves` is better than `user.promoted.user_promote_user`, let me change it!